tests/end2end/model.py

```diff
@@ -63,6 +75,16 @@ def low_rank_update(t, rank):
     return {"A": R, "B": C}
 
 
+def make_ia3_update(value):
+    ia3 = np.random.randn(*value.shape)
+    if ia3.ndim > 3:
```

Comment:
Probably simpler to have this section be something like

```python
axes = (0, -1) if ia3.ndim > 3 else (-1,)
ia3 = np.mean(ia3, axis=axes, keepdims=True)
```

Reply:
Thanks for your suggestion, I made the modification. 

Can you help me with the following?

The below code in test.sh breaks for IA3 
```
echo "Verify that 'git status' doesn't have a diff."
git diff-index --quiet HEAD --
if [[ "$?" != 0 ]]; then
    exit 1
fi
```
The problem is that `git status` shows the file model.pt as modified after checking out to main in the test.sh file. I am not sure how to fix this issue.

Also, I noticed that if I remove the --quiet argument, we can't catch the error and the test.sh file says that the IA3 test is passed. Do you have any suggestions on how to handle this?

Reply:
See https://github.com/r-three/git-theta/pull/185/files#r1169194198 on how to fix it.

Don't remove the `--quiet` argument, it's there so we can look at the exit code to see if it failed lol

Reply:
hahaha, I removed to see the error message. But it turns out removing `--quiet` escapes the if condition :)